scipy/stats/_hypotests.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import division, print_function, absolute_import
+import numpy as np
+import warnings
+from .stats import iqr
+from ._continuous_distns import chi2
+
+
+def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+    """
+    Compute the Epps-Singleton test statistic.
+
+    This is a test for the null hypothesis that two samples have the same
+    underlying probability distribution which is based on the empirical
+    characteristic function.
+
+    Parameters
+    ----------
+    x, y : array-like
+        The two samples of observations to be tested. Input must not have more
+        than one dimension. Samples can have different lengths.
```

Comment:
>Input must not have more than one dimension.

This method is supposed to handle discrete cases just like continuous something that KS test doesn't do (according to [2]). I can see a use case with lots of data where forcing the x,y to be array like forces you to create a lot more data. For example, x,y discrete, and the numbers 0,1,2,3,4 occur often so you include weights for each so you'd want x, (e.g.) to be [[x_1, x_2],[w_1, w_2]] where x_1 and x_2 are entries from 0,1,2,3,4 and w_1, w_2 are the occurrences/counts.

Reply:
@chrisb83 maybe add a #TODO for the weights enhancement somewhere in the code, or do you think the structural specifics of the test does not allow for weight handling?

Reply:
resolved based on comment https://github.com/scipy/scipy/pull/9539#issuecomment-462575451